### PR TITLE
RPITITs are `DefKind::Opaque` with new lowering strategy

### PR DESCRIPTION
**Pull request by compiler-errors**
r? @spastorino 

Kinda cherry-picked #109400

### CODE REVIEW

File: tests/ui/impl-trait/in-trait/foreign.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // check-pass
-// aux-build: rpitit.rs
+// [next] aux-build: rpitit.rs
+// [current] aux-build: rpitit-next.rs
```
Comment (spastorino) [inline tests/ui/impl-trait/in-trait/foreign.rs]: I think you forgot to push the file?. Why is this needed?

File: tests/ui/impl-trait/in-trait/auxiliary/rpitit-next.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// compile-flags: -Zlower-impl-trait-in-trait-to-assoc-ty
+
+#![feature(return_position_impl_trait_in_trait)]
```
Comment (compiler-errors) [inline tests/ui/impl-trait/in-trait/auxiliary/rpitit-next.rs:3]: @spastorino the file is right here :smile:

Comment (compiler-errors) [reply tests/ui/impl-trait/in-trait/auxiliary/rpitit-next.rs:3]: sorry, it was grouped into the wrong commit.

Comment (compiler-errors) [reply tests/ui/impl-trait/in-trait/auxiliary/rpitit-next.rs:3]: Hold on one second -- testing.

Comment (compiler-errors) [reply tests/ui/impl-trait/in-trait/auxiliary/rpitit-next.rs]: Commented on your PR.

**Comment by spastorino** [comment]
You can probably grab my updated commit that includes your suggestion and we can close that other PR.
